homeassistant/components/lg_thinq/humidifier.py

```diff
@@ -0,0 +1,224 @@
+    """Describes ThinQ humidifier entity."""
+
+    current_humidity_key: str
+    operation_key: str
+    mode_key: str = ThinQProperty.CURRENT_JOB_MODE
+
+
+DEVICE_TYPE_HUM_MAP: dict[DeviceType, ThinQHumidifierEntityDescription] = {
+    DeviceType.DEHUMIDIFIER: ThinQHumidifierEntityDescription(
+        key=ThinQProperty.TARGET_HUMIDITY,
+        name=None,
+        device_class=HumidifierDeviceClass.DEHUMIDIFIER,
+        translation_key="dehumidifier",
+        current_humidity_key=ThinQProperty.CURRENT_HUMIDITY,
+        operation_key=ThinQProperty.DEHUMIDIFIER_OPERATION_MODE,
+    ),
+    DeviceType.HUMIDIFIER: ThinQHumidifierEntityDescription(
+        key=ThinQProperty.TARGET_HUMIDITY,
+        name=None,
+        device_class=HumidifierDeviceClass.HUMIDIFIER,
+        translation_key="humidifier",
+        current_humidity_key=ThinQProperty.HUMIDITY,
+        operation_key=ThinQProperty.HUMIDIFIER_OPERATION_MODE,
+    ),
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up an entry for humidifier platform."""
+    entities: list[ThinQHumidifierEntity] = []
+    for coordinator in entry.runtime_data.coordinators.values():
+        if (
+            description := DEVICE_TYPE_HUM_MAP.get(coordinator.api.device.device_type)
+        ) is not None:
+            entities.extend(
+                ThinQHumidifierEntity(coordinator, description, property_id)
+                for property_id in coordinator.api.get_active_idx(
+                    description.key, ActiveMode.READ_WRITE
+                )
+            )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class ThinQHumidifierEntity(ThinQEntity, HumidifierEntity):
+    """Represent a ThinQ humidifier entity."""
+
+    entity_description: ThinQHumidifierEntityDescription
+    _attr_supported_features = HumidifierEntityFeature.MODES
+
+    def __init__(
+        self,
+        coordinator: DeviceDataUpdateCoordinator,
+        entity_description: ThinQHumidifierEntityDescription,
+        property_id: str,
+    ) -> None:
+        """Initialize a humidifier entity."""
+        super().__init__(coordinator, entity_description, property_id)
+        self._attr_available_modes = self.coordinator.data[
+            self.entity_description.mode_key
+        ].options
+
+        if self.data.max is not None:
+            self._attr_max_humidity = self.data.max
+        if self.data.min is not None:
+            self._attr_min_humidity = self.data.min
+        self._target_humidity_step = self.data.step if self.data.step is not None else 1
+
+    def _update_status(self) -> None:
+        """Update status itself."""
+        super()._update_status()
+
+        self._attr_target_humidity = self.data.value
+        self._attr_current_humidity = self.coordinator.data[
```

Comment:
When the dehumidifier is off and its humidity sensor is set to "Turns on with device" in the ThinQ App, the humidity sensor returns a value of 0% to Home Assistant.
We could correct the value here:
```suggestion
        current_humidity = self.coordinator.data[
            self.entity_description.current_humidity_key
        ].value
        self._attr_current_humidity = current_humidity if current_humidity != 0 else None
```

Reply:
@MathiasChaouche 
The model information I have doesn't include the "Turns on with device" sensor attribute. Please let me know your device information and I'll check with the person in charge.
1. Product name, 2. Device info in HA's device